org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.java

```diff
@@ -0,0 +1,489 @@
+		}
+		if (isSelected(preferredBuildSystem, BuildSystem.GRADLE) && createUiProject.getSelection()) {
+			reportIssue(WARNING, Messages.AdvancedNewProjectPage_eclipseAndGradleWarn);
+		}
+		if (createUiProject.getSelection() && createP2Project.getSelection() && !createSDKProject.getSelection()) {
+			addIssue(INFORMATION, Messages.AdvancedNewProjectPage_p2AndSdkInfo);
+		}
+
+		Object source = null;
+		if (e != null) {
+			source = e.getSource();
+		}
+
+		if (createUiProject.getSelection() && !isSelected(sourceLayout, SourceLayout.PLAIN)) {
+			if (createUiProject == source) {
+				String message = "'" + createUiProject.getText() + "' requires " + SourceLayout.PLAIN + " source layout. " //
+						+ "Please <a>select '" + SourceLayout.PLAIN + "'</a> source layout.";
+				reportIssue(ERROR, message, () -> select(sourceLayout, SourceLayout.PLAIN));
+			} else {
+				String message = SourceLayout.MAVEN + " source layout is not supported by the '" + createUiProject.getText() + "' project. " //
+						+ "Please <a>deselect '" + createUiProject.getText() + "'</a>.";
+				reportIssue(ERROR, message, () -> createUiProject.setSelection(false));
+			}
+		}
+
+		if (isSelected(preferredBuildSystem, BuildSystem.NONE) && isSelected(sourceLayout, SourceLayout.MAVEN)) {
+			if (preferredBuildSystem == source) {
+				String message = "Maven/Gradle source layout is not supported without a build system. " //
+						+ "Please <a>select '" + SourceLayout.PLAIN + "'</a> source layout.";
+				reportIssue(ERROR, message, () -> select(sourceLayout, SourceLayout.PLAIN));
+			} else {
+				String message = "Maven/Gradle source layout is only supported when using Maven or Gradle build system. " //
+						+ "You need to choose Maven or Gradle build system. Select <a>Gradle</a> build.";
+				reportIssue(ERROR, message, () -> select(preferredBuildSystem, BuildSystem.GRADLE));
+			}
+		}
+
+		if (createWebProject.getSelection() && isSelected(preferredBuildSystem, BuildSystem.NONE)) {
+			if (preferredBuildSystem == source) {
+				String message = "The '" + createWebProject.getText() + "' project can not be build without a build system. " //
+						+ "Please <a>deselect '" + createWebProject.getText() + "'</a>.";
+				reportIssue(ERROR, message, () -> createWebProject.setSelection(false));
+			} else {
+				String message = "To build the '" + createWebProject.getText() + "' " //
+						+ "project, you need to choose Maven or Gradle build system." //
+						+ "Select <a>Gradle</a> build.";
+				reportIssue(ERROR, message, () -> select(preferredBuildSystem, BuildSystem.GRADLE));
+			}
+		}
+
+		boolean isSelected = Iterables.any(uiButtons, uiButton -> uiButton.getSelection());
+		if (isSelected) {
+			createIdeProject.setEnabled(false);
+		}
+
+		if (autoSelectIdeProject) {
+			autoSelectIdeProject = false;
+			String message = "'" + createIdeProject.getText() + "' project was automatically selected as option '"
+					+ ((Button) source).getText() + "' requires it.";
```

Comment:
@cdietrich @szarnekow @tivervac The value of `source` can possible be `null` here. It is references at multiple points in this code, however, it might fail here. I can add safety check code, however, it is not clear to me where to add it. Please suggest.

Reply:
After a bit of debugging, I learned that the control would come in the current if block and the following one only if the value of `autoSelectIdeProject` and `autoSelectSDKProject` is true. For this to happen, you have to select either “Eclipse Plug-in” or “Create update site” option, in which case the value of the `source` object won't be null.

I think it is ok if we don't add any safety check for the `source` object.